controllers/mattermost/mattermost/mattermost.go

```diff
@@ -103,6 +103,13 @@ func (r *MattermostReconciler) checkMattermostService(
 		return err
 	}
 
+	if current.Spec.Type != desired.Spec.Type && current.Spec.Type != "" {
+		err := r.Resources.DeleteService(types.NamespacedName{Namespace: current.Namespace, Name: current.Name}, reqLogger)
+		if err != nil {
+			return errors.Wrap(err, "failed to delete service")
+		}
+	}
```

Comment:
We are trying to update the service after the deletion, which will always fail. This probably works due to the CR being requeued for reconciliation but produces and error log for the behaviour that is expected.

I think we could just attempt to recreate the Service and return early, since we do not need to copy auto assigned fields as the Service is recreated anyway.
```suggestion
	if current.Spec.Type != desired.Spec.Type && current.Spec.Type != "" {
		err := r.Resources.DeleteService(types.NamespacedName{Namespace: current.Namespace, Name: current.Name}, reqLogger)
		if err != nil {
			return errors.Wrap(err, "failed to delete service")
		}
		return r.Resources.Create(mattermost, desired, reqLogger)
	}
```